pipeline/scraper/test.py

```diff
@@ -1,16 +1,115 @@
-# This is used to test PSQL, ignore this
-#!/usr/bin/python
-import psycopg2
+# Test Script
+# To run test, use the command under the main folder
+# DB_USER=postgres DB_PASSWORD=secret DB_HOST=localhost TOKEN=<token> python3 -m unittest scraper/test.py -v in the scraper folder
+
 import sys
+sys.path.append('./scraper')
+
 import PSQL
- 
-def main():
-    # db = PSQL.connectToDB()
-    # cur = db.cursor()
-    # cur.execute("SELECT * FROM applications;")
-    # result = cur.fetchall()
-    # print(result)
-    print(type({}))
+import GitHubQuery
+import unittest
+import json
+import urllib
+import datetime
+
+def make_orderer():
+    order = {}
+
+    def ordered(f):
+        order[f.__name__] = len(order)
+        return f
+
+    def compare(a, b):
+        return [1, -1][order[a] < order[b]]
+
+    return ordered, compare
+
+ordered, compare = make_orderer()
+unittest.defaultTestLoader.sortTestMethodsUsing = compare
+
+class TestMyClass(unittest.TestCase):
+
+    @ordered
+    def test_runQuery(self):
+        return
+    
+
+    @ordered
+    def test_runQueryOnce(self):
+        monthStr = "created:2020-01-01..2020-02-01"
+        cursor = "Y3Vyc29yOjE="
+        for i in [1,10,100]: 
+            result = GitHubQuery.runQueryOnce(i, monthStr, cursor)
+            json_obj = None
+            try:
+                json_obj = json.load(result)
+                print(json_obj)
+                self.assertTrue(json_obj is not None)
+            except:
+                self.assertFalse(json_obj is not None)
+    
+
+    @ordered
+    def test_connectToDB(self):
+        db = PSQL.connectToDB()
+        self.assertTrue(db is not None)
+
+
+    @ordered
+    def test_insertToApplication(self):
+        db = PSQL.connectToDB()
+        url = "www.pkgpkr.com"
+        followers = 314
+        appName = "pkgpkr"
+        myHash = hash(appName)
+        id = PSQL.insertToApplication(db,url,followers,appName,myHash)
```

Comment:
We should clean up these database entries after the test is complete, otherwise on subsequent runs you're not inserting new records.

Reply:
This is unit test will not leave anything in the database.